File: page.go
package main

import "fmt"

type Student struct {
	ID     int
	Name   string
	Age    int
	Marks  float64
}

var students []Student

func addStudent() {
	var s Student

	fmt.Print("Enter ID: ")
	fmt.Scan(&s.ID)

	fmt.Print("Enter Name: ")
	fmt.Scan(&s.Name)

	fmt.Print("Enter Age: ")
	fmt.Scan(&s.Age)

	fmt.Print("Enter Marks: ")
	fmt.Scan(&s.Marks)

	students = append(students, s)

	fmt.Println("Student added")
}

func displayStudents() {

	fmt.Println("Student Records")

	for _, s := range students {
		fmt.Println("----------------")
		fmt.Println("ID:", s.ID)
		fmt.Println("Name:", s.Name)
		fmt.Println("Age:", s.Age)
		fmt.Println("Marks:", s.Marks)
	}
}

func searchStudent() {

	var id int

	fmt.Print("Enter ID: ")
	fmt.Scan(&id)

	for _, s := range students {

		if s.ID == id {
			fmt.Println("Student Found")
			fmt.Println(s.Name)
			return
		}
	}

	fmt.Println("Student not found")
}

func main() {

	var choice int

	for {

		fmt.Println("\nStudent Management")
		fmt.Println("1. Add")
		fmt.Println("2. Display")
		fmt.Println("3. Search")
		fmt.Println("4. Exit")

		fmt.Print("Choice: ")
		fmt.Scan(&choice)

		switch choice {

		case 1:
			addStudent()

		case 2:
			displayStudents()

		case 3:
			searchStudent()

		case 4:
			fmt.Println("Goodbye")
			return

		default:
			fmt.Println("Invalid choice")
		}
	}
}

File: page.kt
data class Student(
    val id: Int,
    val name: String,
    val age: Int,
    val marks: Double
) {

    fun display() {
        println("---------------------")
        println("ID: $id")
        println("Name: $name")
        println("Age: $age")
        println("Marks: $marks")
    }
}


val students = mutableListOf<Student>()



fun addStudent() {

    print("Enter Student ID: ")
    val id = readLine()!!.toInt()


    print("Enter Student Name: ")
    val name = readLine()!!


    print("Enter Student Age: ")
    val age = readLine()!!.toInt()


    print("Enter Student Marks: ")
    val marks = readLine()!!.toDouble()



    val student = Student(
        id,
        name,
        age,
        marks
    )


    students.add(student)


    println("Student added successfully")
}





fun displayStudents() {

    println("Student Records")


    if (students.isEmpty()) {

        println("No students available")
        return

    }


    for (student in students) {

        student.display()

    }

}





fun searchStudent() {

    print("Enter Student ID: ")
    val id = readLine()!!.toInt()



    val student = students.find {
        it.id == id
    }



    if (student != null) {

        println("Student Found")
        student.display()

    }
    else {

        println("Student not found")

    }

}





fun deleteStudent() {

    print("Enter Student ID: ")
    val id = readLine()!!.toInt()



    val removed = students.removeIf {
        it.id == id
    }



    if (removed) {

        println("Student deleted")

    }
    else {

        println("Student not found")

    }

}





fun calculateAverage() {

    if (students.isEmpty()) {

        println("No data available")
        return

    }


    val average =
        students.map {
            it.marks
        }.average()



    println(
        "Average Marks: $average"
    )

}





fun main() {


    var choice: Int



    do {


        println()
        println("===== Student Management System =====")
        println("1. Add Student")
        println("2. Display Students")
        println("3. Search Student")
        println("4. Delete Student")
        println("5. Calculate Average")
        println("6. Exit")



        print("Enter choice: ")

        choice = readLine()!!.toInt()



        when(choice) {


            1 -> addStudent()


            2 -> displayStudents()


            3 -> searchStudent()


            4 -> deleteStudent()


            5 -> calculateAverage()


            6 -> println("Program ended")


            else -> println("Invalid choice")

        }



    } while(choice != 6)

}
